scripts/glean.py

```diff
@@ -196,6 +196,31 @@ def get_dependencies(self):
         logging.info(f"For {self.app_id}, found Glean dependencies: {dependencies}")
         return dependencies
 
+    def deduplicate_metrics(self, metrics):
+        metric_names = set()
+        results = []
+
+        def get_index(name, list):
+            for i in range(len(list)):
+                if list[i][0] == name:
+                    return i
+            return None
+
+        def get_last_date(metric):
+            return metric[1]["history"][-1]["dates"]["last"]
+
+        for item in metrics:
+            if item[0] not in metric_names:
+                metric_names.add(item[0])
+                results.append(item)
+            # get the latest version of a metric if there's a duplicate
+            if item[0] in metric_names and get_last_date(item) > get_last_date(
+                results[get_index(item[0], results)]
+            ):
+                del results[get_index(item[0], results)]
+                results.append(item)
+        return results
```

Comment:
This looks the right idea, however I'd suggest a slightly different implementation which I think will be both shorter and faster: basically create a dictionary from metric names -> metrics iteratively, then just get the values at the end. Here's a simple example of the approach:

```py
x = [
    {
    'id': 1,
    'date': 1
    },
    {
    'id': 1,
    'date': 2
    },
    {
    'id': 1,
    'date': 0
    }
]
y = {}
for x1 in x:
    if not y.get(x1['id']) or y[x1['id']]['date'] < x1['date']:
            y[x1['id']] = x1

y.values()    
```
https://alpha.iodide.io/notebooks/7408/

Reply:
this is a much simpler approach! thank you!